snuba/datasets/discover.py

```diff
@@ -306,33 +306,13 @@ def column_expr(
         if detected_dataset == TRANSACTIONS:
             if column_name == "time":
                 return self.time_expr("finish_ts", query.get_granularity(), table_alias)
-            if column_name == "type":
-                return "'transaction'"
```

Comment:
I'm not sure if this one should have actually been moved or not.

Reply:
I think so, the transactions dataset does not have a type column, so if a query needs to run on that dataset the `type` column needs to be resolved.

Reply:
My thinking here — if the query is being directed to transactions, it's kind of a given that the row being returned is a transaction, right? Having to disambiguate between a transaction and any other type of event is really only a concern for discover (since it is the only thing smashing multiple types together into a single result set) so it might be more appropriate there.

It doesn't really _harm_ anything by being handled in transactions instead of events, though. Just potentially unnecessary.

Reply:
Sure, it is unnecessary for the transactions dataset.
The scope of my question depends on how queries for transactions subscriptions will be built in sentry. If they are building standard discover queries and then we will be running them on the transactions datasets, then those query will have a `event.type` condition, that would not have a corresponding field in the transactions logical schema.
But if the query is built directly according to the transactions dataset schema, there will be no type condition (sorry cannot find anymore the document where we discussed the plan to introduce transactions subscription, so I cannot remember how the query will be built).

Anyway not blocking, this can be easily moved back and forth if/when needed

Reply:
Ok, that makes sense. For subscriptions, there will be an explicitly required selection (radio button or dropdown) between errors and transactions (maybe security too) that is pretty much directly mappable to dataset selection (or a `FROM` clause in an entity world.) It's fine to keep it around to simplify copying over a query directly from discover into metric alerts, but I think we might want to consider changing this behavior later.